c7n/resources/sns.py

```diff
@@ -119,3 +119,81 @@ def process_resource(self, client, resource):
         return {'Name': resource['TopicArn'],
                 'State': 'PolicyRemoved',
                 'Statements': found}
+
+
+@SNS.action_registry.register('modify-statements')
+class ModifyPolicyStatement(ModifyPolicyBase):
+    """Action to modify policy statements from SNS
+
+    :example:
+
+    .. code-block:: yaml
+
+           policies:
+              - name: sns-cross-account
+                resource: sns
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: modify-statements
+                    add-statements: [statement]
+                    remove-statements: [statement] or *
+    """
+
+    permissions = ('sns:SetTopicAttributes', 'sns:GetTopicAttributes')
+
+    def process(self, resources):
+        replace = False
+        results = []
+        client = local_session(self.manager.session_factory).client('sns')
+        additions = self.data.get('add-statements', [])
```

Comment:
both `additions` and `deletions` should be able to default to `None` to simply/clarify lines 162, 164, and 166 so they can be 

```python
if additions and deletions and not replace:
```

etc.